pkg/controller/v1beta1/inferenceservice/utils/utils_test.go

```diff
@@ -1719,3 +1721,85 @@ func TestGetPredictorEndpoint(t *testing.T) {
 		})
 	}
 }
+
+func TestValidateStorageURIForDefaultStorageInitializer(t *testing.T) {
+	validUris := []string{
+		"https://kfserving.blob.core.windows.net/triton/simple_string/",
+		"https://kfserving.blob.core.windows.net/triton/simple_string",
+		"https://kfserving.blob.core.windows.net/triton/",
+		"https://kfserving.blob.core.windows.net/triton",
+		"https://raw.githubusercontent.com/someOrg/someRepo/model.tar.gz",
+		"http://raw.githubusercontent.com/someOrg/someRepo/model.tar.gz",
+		"hdfs://",
+		"webhdfs://",
+		"some/relative/path",
+		"/",
+		"foo",
+		"",
+	}
+	s := runtime.NewScheme()
+	err := v1alpha1.AddToScheme(s)
+	if err != nil {
+		klog.Error(err, "Failed to add v1alpha1 to scheme")
+	}
+	mockClient := fake.NewClientBuilder().WithScheme(s).Build()
+	for _, uri := range validUris {
+		if err := ValidateStorageURI(&uri, mockClient); err != nil {
+			t.Errorf("%q validation failed: %s", uri, err)
+		}
+	}
+}
+
+func TestValidateStorageURIForCustomPrefix(t *testing.T) {
+	invalidUris := []string{
+		"custom://custom.com/model",
+	}
+	s := runtime.NewScheme()
+	err := v1alpha1.AddToScheme(s)
+	if err != nil {
+		klog.Error(err, "Failed to add v1alpha1 to scheme")
+	}
+	mockClient := fake.NewClientBuilder().WithScheme(s).Build()
+	for _, uri := range invalidUris {
+		if err := ValidateStorageURI(&uri, mockClient); err == nil {
+			t.Errorf("%q validation failed: error expected", uri)
+		}
+	}
+}
+
+func TestValidateStorageURIForDefaultStorageInitializerCRD(t *testing.T) {
+	customSpec := v1alpha1.ClusterStorageContainer{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: "custom",
+		},
+		Spec: v1alpha1.StorageContainerSpec{
+			Container: v1.Container{
+				Image: "kserve/storage-initializer:latest",
+				Resources: v1.ResourceRequirements{
+					Limits: v1.ResourceList{
+						v1.ResourceMemory: resource.MustParse("200Mi"),
+					},
+				},
+			},
+			SupportedUriFormats: []v1alpha1.SupportedUriFormat{{Prefix: "custom://"}},
+		},
+	}
+
+	storageContainerSpecs := &v1alpha1.ClusterStorageContainerList{
+		Items: []v1alpha1.ClusterStorageContainer{customSpec},
+	}
+	validUris := []string{
+		"custom://custom.com/model/",
+	}
+	s := runtime.NewScheme()
+	err := v1alpha1.AddToScheme(s)
+	if err != nil {
+		klog.Error(err, "Failed to add v1alpha1 to scheme")
```

Comment:
This is part of the test cases. If the error is not expected, we can just fail the tests here. No need to use klog.